src/workerd/jsg/value.h

```diff
@@ -470,14 +471,11 @@ class StringWrapper {
     v8::Local<v8::String> str = check(handle->ToString(context));
     auto result =
         ByteString(KJ_ASSERT_NONNULL(tryUnwrap(context, str, (kj::String*)nullptr, parentObject)));
-    if (str->ContainsOnlyOneByte()) {
+    if (str->ContainsOnlyOneByte()) [[likely]] {
```

Comment:
Wait, this whole block looks suspect. We already decoded the string content to a kj::String... shouldn't we scan that, instead of allocating a whole second buffer and using `WriteOneByte()` to copy the content again?

Also I note the doc comment for `ContainsOnlyOneByte()` says that it "Will read the entire string in some cases." -- in what cases? Can we avoid that scan as well?

Reply:
Yes, you're right. I've changed the code a little bit to avoid any unnecessary performance impact.